sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -251,14 +259,10 @@ Bigquery getClient() {
     @Override
     public void startLoadJob(JobReference jobRef, JobConfigurationLoad loadConfig)
         throws InterruptedException, IOException {
-      Map<String, String> labelMap = new HashMap<>();
       Job job =
           new Job()
               .setJobReference(jobRef)
-              .setConfiguration(
-                  new JobConfiguration()
-                      .setLoad(loadConfig)
-                      .setLabels(this.bqIOMetadata.addAdditionalJobLabels(labelMap)));
```

Comment:
addAdditionalJobLabels will add beam job id which becomes available at some later stage, and now it is called much earlier. Not sure if this is a concern, but can we minimize the scope of the change such that just replace labelMap to this.jobLabels here?

Reply:
Done